// big-command/lib/command/logger.js
var levels = {
  info: 'green',
  data: 'grey',
  warn: 'yellow',
  error: 'red',
  help: 'cyan'
};

var logger = exports;

logger.log = function (level, data) {
  console.log(level[levels[level]] + ':', data);
}

function hoistLevels (levels) {
  Object.keys(levels).forEach(function(level){
    logger[level] = function (data) {
      logger.log(level, data);
    }
  });
};


function pad (str, count) {
  for (var i = 0; i < count; i++) {
    str += " ";
  }
  return str;
};

logger.put = function (input) {
  //
  // is this an array?
  //
  if (Array.isArray(input)) {
    //
    // headers
    //
    var headers = [];
    Object.keys(input[0]).forEach(function(prop){
      headers.push(prop);
    })
    logger.data(headers.join(' '));
    //
    // rows
    //
    input.forEach(function(item){
      logger.put(item);
    });
    return;
  }
  // else, is this an object?
  if (typeof input === 'object') {
    logger.data('{');
    Object.keys(input).forEach(function(prop){
      logger.data('  ' + prop + ': ' + input[prop]);
    });
    logger.data('}');
    return;
  }
  //
  // log as string or number
  //
  logger.data(input);
};

hoistLevels(levels);
